lib/backend/k8s/syncer.go

```diff
@@ -325,9 +354,9 @@ func (syn *kubeSyncer) performSnapshot() ([]model.KVPair, map[model.Key]bool, re
 			labels.Revision = profile.Revision
 
 			snap = append(snap, *rules, *tags, *labels)
-			keys[rules.Key] = true
-			keys[tags.Key] = true
-			keys[labels.Key] = true
+			keys[fmt.Sprintf("%s", rules.Key)] = true
```

Comment:
If we are stating here that the Key must be representable by a unique string then we should be explicit about that with the Key interface.

You either need to:
-  Add the Stringer interface to the Key interface with a clear comment that the string representation of the Key should be a unique identifier for the object within the cluster.... or
-  or, you could use the defaultPath() as the identifier ... although that is really for KV stores.
-  or add another method ... 

personally I'd plump for solidifying the requirement for each Key type to implement a unique String() representation.